substratevm/src/com.oracle.svm.driver/src/com/oracle/svm/driver/NativeImage.java

```diff
@@ -1203,11 +1204,30 @@ protected int buildImage(List<String> javaArgs, LinkedHashSet<Path> bcp, LinkedH
         }
 
         int exitStatus = 1;
+        File argsFile = null;
         try {
+            if (OS.getCurrent() == OS.WINDOWS) {
+                int totalLength = command.stream().mapToInt(s -> s.length() + 1).sum();
+                if (totalLength >= WINDOWS_MAX_COMMAND_LENGTH) {
+                    argsFile = File.createTempFile("native-image", "args");
+                    argsFile.deleteOnExit();
+                    Files.write(argsFile.toPath(), (Iterable<String>) command.stream().skip(1)::iterator);
+                    List<String> atCommand = new ArrayList<>();
+                    atCommand.add(command.get(0));
+                    atCommand.add("@" + argsFile);
+                    command = atCommand;
+                }
+            }
+            ProcessBuilder pb = new ProcessBuilder();
+            pb.command(command);
             Process p = pb.inheritIO().start();
             exitStatus = p.waitFor();
         } catch (IOException | InterruptedException e) {
             throw showError(e.getMessage());
+        } finally {
+            if (argsFile != null) {
```

Comment:
Use
```
`!useDebugAttach() && argsFile != null`
```
here.

Reply:
I will, but could you explain that to me?

Reply:
I you run `native-image` with `--debug-attach` you would eventually want to inspect the `argsFile` to know exactly what was passed passed to the native-image builder. See
https://github.com/oracle/graal/blob/75df2c1dc9ce10c175261f9d003eb983625a81b6/substratevm/src/com.oracle.svm.driver/src/com/oracle/svm/driver/DefaultOptionHandler.java#L153
https://github.com/oracle/graal/blob/75df2c1dc9ce10c175261f9d003eb983625a81b6/substratevm/src/com.oracle.svm.driver/resources/HelpExtra.txt#L9